sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/GenerateUnsafeProjection.scala

```diff
@@ -318,13 +316,12 @@ object GenerateUnsafeProjection extends CodeGenerator[Seq[Expression], UnsafePro
       case _ => true
     }
 
-    val result = ctx.freshName("result")
-    ctx.addMutableState("UnsafeRow", result, s"$result = new UnsafeRow(${expressions.length});")
+    val result = ctx.addMutableState("UnsafeRow", "result",
+      v => s"$v = new UnsafeRow(${expressions.length});", inline = true)
```

Comment:
ditto

Reply:
We want to keep order of initialization since they have dependency  (`a = new UnsafeRow(...)`, and then `b = new BufferHolder(a, ...);`).

Reply:
Or, is it better to introduce another flag to keep order of initialization even when compacted?  e.g. `keepInitializationOrder`

Reply:
what if we have a single buffer for both inline and array compacted global variable init code?

Reply:
sure